com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/textureset/TextureSetGenerator.java

```diff
@@ -356,7 +379,20 @@ public static TextureSetResult calculateLayout(List<Rect> images, List<SpriteGeo
                 vertexData.left.addGeometries(createPolygonUVs(geometry, rect, layoutWidth, layoutHeight));
             }
         }
-        return new TextureSetResult(vertexData.left, vertexData.right, new LayoutResult(layouts, innerPadding, extrudeBorders));
+
+        TimeProfiler.stop();
+        return new TextureSetResult(vertexData.left, vertexData.right, layout);
+    }
+
+    // Deprecated
+    public static TextureSetResult calculateLayout(List<Rect> images, List<SpriteGeometry> imageHulls, int use_geometries,
+                AnimIterator iterator, int margin, int innerPadding, int extrudeBorders,
+               boolean rotate, boolean useTileGrid, Grid gridSize, float maxPageSizeW, float maxPageSizeH) {
+
+        LayoutResult layout = calculateLayoutResult(images, margin, innerPadding, extrudeBorders, rotate,
+                                                    useTileGrid, gridSize, maxPageSizeW, maxPageSizeH);
+
+        return calculateTextureSetResult(layout, imageHulls, use_geometries, iterator);
```

Comment:
This is the current interface the editor uses, and it shows how to split the calls into two.